src/gt4py/next/ffront/foast_passes/type_alias_replacement.py

```diff
@@ -0,0 +1,99 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from dataclasses import dataclass
+from typing import Any
+
+import gt4py.next.ffront.field_operator_ast as foast
+from gt4py.eve import NodeTranslator, traits
+from gt4py.eve.concepts import SourceLocation, SymbolName, SymbolRef
+from gt4py.next.ffront import dialect_ast_enums
+from gt4py.next.ffront.fbuiltins import TYPE_BUILTIN_NAMES
+from gt4py.next.type_system import type_specifications as ts
+from gt4py.next.type_system.type_translation import get_scalar_kind
+
+
+@dataclass
+class TypeAliasReplacement(NodeTranslator, traits.VisitorWithSymbolTableTrait):
+    """
+    Replace Type Aliases with their actual type.
+
+    After this pass, the type aliases used for explicit construction of literal
+    values and for casting field values are replaced by their actual types.
+    """
+
+    closure_vars: dict[str, Any]
+
+    @classmethod
+    def apply(
+        cls, node: foast.FunctionDefinition | foast.FieldOperator, closure_vars: dict[str, Any]
+    ) -> tuple[foast.FunctionDefinition, dict[str, Any]]:
+        foast_node = cls(closure_vars=closure_vars).visit(node)
+        closures_copy = closure_vars.copy()
+        for key, value in closures_copy.items():
+            if isinstance(value, type) and key not in TYPE_BUILTIN_NAMES:
+                closure_vars[value.__name__] = closure_vars.pop(key)
+
+        return foast_node, closure_vars
```

Comment:
I modified the above suggested code so that it works (iterating over and returning an empty `new_closure_vars ` dict) and paid attention not to modify the `closure_vars `arg. Here is the corrected loop: If this was not the intended change, please let me know.
```
        new_closure_vars = closure_vars.copy()
        for key, value in closure_vars.items():
            if isinstance(value, type) and key not in TYPE_BUILTIN_NAMES:
                new_closure_vars[value.__name__] = closure_vars[key]
        return foast_node, new_closure_vars

```